test/test_optim.py

```diff
@@ -112,7 +112,15 @@ def closure():
 
             initial_value = closure().item()
             for _ in range(20):
-                loss = optimizer.step(closure)
+                if torch._dynamo.is_compiling() and optim_cls != torch.optim.LBFGS:
```

Comment:
Use the optim_info.step_requires_closure flag instead. Can you add a separate test to test correctness of closures under torch.compile in test_compiled_optimizers.py (that it falls back and computes the same thing)?